### DISCUSSION TITLE
Adding Table customConstraints appears to hang moor

### DISCUSSION THREAD

**Discussion by IanWorthington** [simolus3/drift#1173] [Q&A]
I have a class:

```dart
class Instruments extends Table { 
  IntColumn get id => integer().autoIncrement()();
  IntColumn get dataSource => intEnum<DataSource>()();
  TextColumn get symbol => text().withLength(min: 1)();
  ...

  @override
  List<String> get customConstraints => [
    'UNIQUE (dataSource, symbol)'
  ];
}
```

When the customContraints getter is commented out, all is well.  Enabling it appears to cause something to hang, maybe during database creation?  I have a migration which deletes and recreates all tables.  Running it without the customConstraints getter logs that plus the first get:

```
I/flutter (11753): Moor: Sent DROP TABLE IF EXISTS instruments; with args []
I/flutter (11753): Moor: Sent CREATE TABLE IF NOT EXISTS instruments (id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, last_updated INTEGER NOT NULL DEFAULT (strftime('%s', CURRENT_TIMESTAMP)), data_source INTEGER NOT NULL, symbol VARCHAR NOT NULL, name VARCHAR NOT NULL, exchange VARCHAR NOT NULL, type VARCHAR NOT NULL); with args []
I/flutter (11753): Database table instruments has been deleted and recreated.
I/flutter (11753): Moor: Sent SELECT * FROM instruments; with args []
```

However WITH the getter in place I get only:

```
I/flutter (11496): Moor: Sent DROP TABLE IF EXISTS instruments; with args []
I/flutter (11496): Moor: Sent CREATE TABLE IF NOT EXISTS instruments (id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, last_updated INTEGER NOT NULL DEFAULT (strftime('%s', CURRENT_TIMESTAMP)), data_source INTEGER NOT NULL, symbol VARCHAR NOT NULL, name VARCHAR NOT NULL, exchange VARCHAR NOT NULL, type VARCHAR NOT NULL, UNIQUE (dataSource, symbol)); with args []
```

ie no complete message nor  the first get.

Have I done something wrong here?

**Comment by simolus3** [comment]
Running the second  `CREATE TABLE` statement causes an error: "no such column: dataSource". This is because moor transforms column names to `camel_case` by default, so you should either

- declare the constraint as `UNIQUE(data_source, symbol)`
- Add a custom name: `IntColumn get dataSource => intEnum<DataSource>().named('dataSource')();`

> ie no complete message nor the first get.

Moor should have thrown an exception in that case, do you see that anywhere?
